plotly/tools.py

```diff
@@ -1000,8 +1000,7 @@ def _get_anchors(r, c, x_cnt, y_cnt, shared_xaxes, shared_yaxes):
     # Function pasting x/y domains in layout object (2d case)
     def _add_domain(layout, x_or_y, label, domain, anchor, position):
         name = label[0] + 'axis' + label[1:]
-        graph_obj = '{X_or_Y}Axis'.format(X_or_Y=x_or_y.upper())
-        axis = getattr(graph_objs, graph_obj)(domain=domain)
+        axis = {'domain': domain}
```

Comment:
I suppose you did this because `axis` will never be equal to anything other than `{'domain': domain}`, and so it's like a trivial case of memoization? :)